lbz/jwt_utils.py

```diff
@@ -54,29 +41,32 @@ def validate_jwt_properties(decoded_jwt: dict) -> None:
     if "iss" not in decoded_jwt:
         raise SecurityError("The auth token doesn't have the 'iss' field.")
     issuer = decoded_jwt["iss"]
-    if not issuer or issuer not in ALLOWED_ISS:
+    if not issuer or issuer not in ALLOWED_ISS.value:
         raise Unauthorized(f"{issuer} is not an allowed token issuer")
 
 
 def decode_jwt(auth_jwt_token: str) -> dict:  # noqa:C901
     """
     Decodes JWT token.
     """
-    if not PUBLIC_KEYS:
-        msg = "Invalid configuration - no keys in the ALLOWED_PUBLIC_KEYS env variable"
-        raise RuntimeError(msg)
 
-    if not ALLOWED_AUDIENCES:
-        raise RuntimeError("ALLOWED_AUDIENCES were not defined")
+    if not ALLOWED_PUBLIC_KEYS.value:
+        raise MissingConfigValue("ALLOWED_PUBLIC_KEYS")
+
+    if not ALLOWED_AUDIENCES.value:
+        raise MissingConfigValue("ALLOWED_AUDIENCES")
+
+    if any("kid" not in public_key for public_key in ALLOWED_PUBLIC_KEYS.value):
```

Comment:
This line checks the correctness of the environment variable, so, I think it should be implemented directly in the parser: `ConfigParser.load_jwt_keys()` 😉